lib/data_update_scripts/20211104200856_update_each_articles_privileged_user_point_count.rb

```diff
@@ -0,0 +1,17 @@
+module DataUpdateScripts
+  class UpdateEachArticlesPrivilegedUserPointCount
+    def run
+      # This code is "idempotent" in that it's calculating and caching a value stored elsewhere.
+      Article.find_in_batches do |articles|
+        articles.each do |article|
+          # Given that we have a default of 0, don't update if we
+          # don't have any reactions.
+          next unless article.reactions.privileged_category.exists?
+
+          article.update_column(:privileged_users_reaction_points_sum,
+                                article.reactions.privileged_category.sum(:points))
+        end
+      end
```

Comment:
I'm only slightly worried that this whole thing might take too long and timeout. If that does happen, what we did that to resolve in the past is to have each update queue up as its own job (see [this deleted file](https://github.com/forem/forem/blob/2438cd33bc6cbda206f5666e104d00fad05db708/lib/data_update_scripts/20210423155327_migrate_relevant_fields_from_users_to_users_settings.rb))

I don't think this will happen but whenever we have to run anything in `Article`, I want to take that into account. So I'd say LGTM and revisit if there's an issue. Can I get @jgaskins to gut check this?

Reply:
Interesting…definitely want to hear @jgaskins thoughts?

I could go directly to SQL instead? (Though I'll need to remember a bit of those antics)

Reply:
@maestromac Since this is running in multiple queries, it should be okay. The timeout is per SQL statement and is set on the connection with `SET statement_timeout = X` (configured in [`database.yml`](https://github.com/forem/forem/blob/bff7cd11181e53358584eb312580b1a8587fd31a/config/database.yml#L26) and set to 10 seconds on DEV).

> I could go directly to SQL instead? (Though I'll need to remember a bit of those antics)

@jeremyf Since this timeout is enforced by the Postgres server (rather than within ActiveRecord[1]) on all connections spun up by ActiveRecord, raw SQL wouldn't avoid the timeout. If we need to avoid the timeout, we would need to extend it using [a `ApplicationRecord#with_statement_timeout` block](https://github.com/forem/forem/blob/bff7cd11181e53358584eb312580b1a8587fd31a/app/models/application_record.rb#L63-L71) ([example usage](https://github.com/forem/forem/pull/15085/files#diff-df7c5d2c9cac7f53f9656ad174b1a2356652e7ff469e09486892f0e0c727533bR42-R58)).

---

[1] This is a good because Ruby `Timeout` could leave the Postgres connection in an invalid state. See https://jvns.ca/blog/2015/11/27/why-rubys-timeout-is-dangerous-and-thread-dot-raise-is-terrifying/